### PR TITLE
Added documentation

### PR DESCRIPTION
**Pull request by dutow**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,5 +1,39 @@
...
+```json
+{
+        'provider': 'file',
+        'datafile': '/tmp/pgkeyring',
+}
+```
+
+Currently the keyring configuration only supports the file provider, with a single datafile parameter.
+
+This setup is intended for developmenet, and stores the keys unencrypted in the specified data file.
```
Comment (dAdAbird) [inline README.md:26]:
```suggestion
This setup is intended for development and stores the keys unencrypted in the specified data file.
```

File: README.md [hunk 2]
```diff
@@ -1,5 +1,39 @@
...
+## Keyring configuration
+
+```json
+{
+        'provider': 'file',
+        'datafile': '/tmp/pgkeyring',
+}
+```
+
+Currently the keyring configuration only supports the file provider, with a single datafile parameter.
```
Comment (dAdAbird) [inline README.md:24]:
```suggestion
Currently, the keyring configuration only supports the file provider with a single datafile parameter.
```

File: README.md [hunk 3]
```diff
@@ -1,5 +1,39 @@
...
+
+This setup is intended for developmenet, and stores the keys unencrypted in the specified data file.
+
+## Build steps
+
+1. Install libjson-c-dev: `apt install libjon-c-dev`
+2. Install or build postgresql 16 (see reference commit below)
+3. If postgres is installed in a non standard directory, set the `PG_CONFIG` environment variable to point to the `pg_config` executable
```
Comment (dAdAbird) [inline README.md:32]:
```suggestion
3. If postgres is installed in a non-standard directory, set the `PG_CONFIG` environment variable to point to the `pg_config` executable
```

**Review by dAdAbird — APPROVED** [review]
LGTM
just minor grammatical suggestions

**Review by capri1989 — CHANGES_REQUESTED** [review]

File: README.md [hunk 4]
```diff
@@ -1,5 +1,39 @@
...
+This is an experimental encrypted access method for Postgres 16.
+
+## Installation steps
+
+1. Build and install the plugin either with make or meson (see build steps)
+2. Add pg_tde to the preload libraries: `ALTER SYSTEM SET shared_preload_libraries = 'pg_tde';`
+3. Restart the postgres server
+4. Create the extension: `CREATE EXTENSION pg_tde;`
+5. Set the location of the keyring configuration file in postgresql.conf: `pg_tde.keyringConfigFile = '/where/to/put/the/keyring.json'`
+6. Create the keyring configuration file (see example keyring configuration)
```
[the remark this answers is not in the document]
Comment (dutow) [reply README.md:12]: There's an example later in the file

[the remark this answers is not in the document]
Comment (capri1989) [reply README.md:12]: ok but what's needed for the datafile? Just an empty file? What needs to be inside this file in which format?

**Review by capri1989 — CHANGES_REQUESTED** [review]

File: README.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (dutow) [reply README.md:12]: That's created and managed by the server - I'll mention this to the documentation.

File: README.md [hunk 5]
```diff
@@ -1,5 +1,39 @@
...
+}
+```
+
+Currently the keyring configuration only supports the file provider, with a single datafile parameter.
+
+This setup is intended for developmenet, and stores the keys unencrypted in the specified data file.
+
+## Build steps
+
+1. Install libjson-c-dev: `apt install libjon-c-dev`
```
[the remark this answers is not in the document]
Comment (dutow) [reply README.md]: I've mentinoed that's an exmaple for ubuntu - as I'm using that, I didn't test other systems.

File: README.md [hunk 6]
```diff
@@ -1,5 +1,39 @@
...
+
+## Installation steps
+
+1. Build and install the plugin either with make or meson (see build steps)
+2. Add pg_tde to the preload libraries: `ALTER SYSTEM SET shared_preload_libraries = 'pg_tde';`
+3. Restart the postgres server
+4. Create the extension: `CREATE EXTENSION pg_tde;`
+5. Set the location of the keyring configuration file in postgresql.conf: `pg_tde.keyringConfigFile = '/where/to/put/the/keyring.json'`
+6. Create the keyring configuration file (see example keyring configuration)
+6. Restart the postgres server
```
[the remark this answers is not in the document]
Comment (dutow) [reply README.md]: Fixed.

**Review by capri1989 — APPROVED** [review]
LGTM

**Review by codeforall — APPROVED** [review]

**Pull request merged**